config/initializers/omniauth.rb

```diff
@@ -40,13 +40,22 @@ def callback_phase
         fail!(:timeout, e)
       rescue ::SocketError => e
         fail!(:failed_to_connect, e)
+      # Adding this provides some nicer / additional logging but not why the error was caused
+      rescue Rack::OAuth2::Client::Error => e
+        fail!(:invalid_client, e)
       end
     end
   end
 end
 
 class OmniAuth::Strategies::Dfe < OmniAuth::Strategies::OpenIDConnect; end
 
+# Removing the default secret from config/settings.yml allows app to fail fast
+# with explicit reason as to why
+if !Settings.dfe_signin.secret.present?
+  raise 'dfe_signin secret not set in config/settings'
```

Comment:
is this just for dev env?

Reply:
It will in any environment currently ...